contrib/cirrus/runner.sh

```diff
@@ -22,62 +22,83 @@ OS_RELEASE_ID="$(source /etc/os-release; echo $ID)"
 OS_RELEASE_VER="$(source /etc/os-release; echo $VERSION_ID | tr -d '.')"
 # Combined to ease some usage
 OS_REL_VER="${OS_RELEASE_ID}-${OS_RELEASE_VER}"
+# This is the magic interpreted by the tests to allow modifying local config/services.
+export SKOPEO_CONTAINER_TESTS=1
 
 export "PATH=$PATH:$GOPATH/bin"
 
-podmanmake() {
-    req_env_vars GOPATH SKOPEO_PATH SKOPEO_CI_CONTAINER_FQIN
-    warn "Accumulated technical-debt requires execution inside a --privileged container.  This is very likely hiding bugs!"
-    showrun podman run -it --rm --privileged \
-        -e GOPATH=$GOPATH \
-        -v $GOPATH:$GOPATH:Z \
-        -w $SKOPEO_PATH \
-        $SKOPEO_CI_CONTAINER_FQIN \
-            make "$@"
-}
-
 _run_setup() {
+    local mnt
+    local errmsg
+    req_env_vars SKOPEO_CI_CONTAINER_FQIN
     if [[ "$OS_RELEASE_ID" == "fedora" ]]; then
         # This is required as part of the standard Fedora VM setup
         growpart /dev/sda 1
         resize2fs /dev/sda1
 
-        # VM's come with the distro. skopeo pre-installed
+        # VM's come with the distro. skopeo package pre-installed
         dnf erase -y skopeo
     else
         die "Unknown/unsupported distro. $OS_REL_VER"
     fi
 }
 
 _run_vendor() {
-    podmanmake vendor BUILDTAGS="$BUILDTAGS"
+    make vendor BUILDTAGS="$BUILDTAGS"
 }
 
 _run_build() {
     make bin/skopeo BUILDTAGS="$BUILDTAGS"
 }
 
 _run_cross() {
-    podmanmake local-cross BUILDTAGS="$BUILDTAGS"
+    make local-cross BUILDTAGS="$BUILDTAGS"
 }
 
 _run_doccheck() {
     make validate-docs BUILDTAGS="$BUILDTAGS"
 }
 
 _run_unit() {
-    podmanmake test-unit-local BUILDTAGS="$BUILDTAGS"
+    make test-unit-local BUILDTAGS="$BUILDTAGS"
 }
 
 _run_integration() {
-    podmanmake test-integration-local BUILDTAGS="$BUILDTAGS"
+    # A slew of compiled binaries are pre-built and distributed
+    # within the CI/Dev container image, but we want to run
+    # things directly on the host VM.  Fortunately they're all
+    # located in the container under /usr/local/bin
+    msg "Accessing contents of $SKOPEO_CI_CONTAINER_FQIN"
+    podman pull --quiet $SKOPEO_CI_CONTAINER_FQIN
+    mnt=$(podman mount $(podman create $SKOPEO_CI_CONTAINER_FQIN))
+
+    # This should never ever ever happen, so double-check to make sure.
+    errmsg="Somehow $SKOPEO_CI_CONTAINER_FQIN is not based on $OS_REL_VER."
+    if fgrep -qx "ID=$OS_RELEASE_ID" $mnt/etc/os-release; then
+        if ! fgrep -qx "VERSION_ID=$OS_RELEASE_VER" $mnt/etc/os-release; then
+            die "$errmsg"
+        fi
+    else
+        die "$errmsg"
+    fi
+    msg "Copying test binaries from $SKOPEO_CI_CONTAINER_FQIN /usr/local/bin/"
+    podman pull --quiet $SKOPEO_CI_CONTAINER_FQIN
```

Comment:
This was already done a bit above; does this do anything extra?

Reply:
Nope, must be leftover. I'll yank it.